examples/hotplug_events.rs

```diff
@@ -0,0 +1,27 @@
+extern crate libusb;
+
+use std::error::Error;
+use std::slice;
+use std::str::FromStr;
+use std::time::Duration;
+
+#[derive(Debug)]
+struct Endpoint {
+    config: u8,
+    iface: u8,
+    setting: u8,
+    address: u8
+}
+
+fn main() -> Result<(), Box<dyn Error>> {
+    let mut ctx = libusb::Context::new()?;
+    ctx.register_callback(Default::default(), |device, event| {
```

Comment:
@dcuddeback This is the user facing API, I'd love your thoughts on this part. I create a native &Device for the callback, and created an Event enum to mirror the libusb side but I'm open to other ideas here.

The libusb interface for filters doesn't really map onto rust super well so I made a builder style thing to let you create filters.